perceiver_pytorch/encoders.py

```diff
@@ -0,0 +1,353 @@
+import torch
+import torch.nn.functional as F
+import numpy as np
+import math
+import einops
+
+def extract_image_patches(x, kernel, stride=1, dilation=1):
```

Comment:
worth putting this function, in utils.py file as maybe it doesnt fi under enconders?

Reply:
Doc string would be great for what i does, , and typing on input variables is very useful

Reply:
I've split up the code into smaller files, and am working on adding typing now!